app/cdap/components/DirectiveInput/utils.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Fuse from 'fuse.js';
+import uuidV4 from 'uuid/v4';
+import { IDirectivesList } from 'components/DirectiveInput/types';
+
+/**
+ * @param  {string} directiveInput
+ * @param  {IDirectivesList[]} directivesList
+ * @returns {array}
+ * most appropriate match is returned from directives list so it can be used to show how the directive is used
+ */
+export const formatDirectiveUsageData = (
+  directiveInput: string,
+  directivesList: IDirectivesList[]
+) => {
+  const inputSplit: string[] = directiveInput.replace(/^\s+/g, '').split(' ');
+  const fuseOptions = {
+    includeScore: true,
+    includeMatches: true,
+    caseSensitive: false,
+    threshold: 0,
+    location: 0,
+    shouldSort: true,
+    distance: 100,
+    minMatchCharLength: 1,
+    maxPatternLength: 32,
+    keys: ['directive'],
+  };
+  const fuse = new Fuse(directivesList, fuseOptions);
+  return fuse.search(inputSplit[0]).map((row) => {
```

Comment:
Why is there a new `Fuse` here? Why would this run a search here?

Reply:
Based on the fuseOptions we are searching in the directivesList.